firebase-firestore/src/test/java/com/google/firebase/firestore/remote/RemoteSerializerTest.java

```diff
@@ -378,64 +378,98 @@ public void testEncodesPatchMutationWithFieldMask() {
   }
 
   @Test
-  public void testEncodesServerTimestampTransformMutation() {
+  public void testEncodesServerTimestampMutation() {
     Mutation mutation =
-        transformMutation(
+        patchMutation(
             "docs/1",
             map(
                 "a",
                 com.google.firebase.firestore.FieldValue.serverTimestamp(),
                 "bar.baz",
                 com.google.firebase.firestore.FieldValue.serverTimestamp()));
-
     Write expected =
         Write.newBuilder()
-            .setTransform(
-                DocumentTransform.newBuilder()
-                    .setDocument("projects/p/databases/d/documents/docs/1")
-                    .addFieldTransforms(
-                        DocumentTransform.FieldTransform.newBuilder()
-                            .setFieldPath("a")
-                            .setSetToServerValue(
-                                DocumentTransform.FieldTransform.ServerValue.REQUEST_TIME))
-                    .addFieldTransforms(
-                        DocumentTransform.FieldTransform.newBuilder()
-                            .setFieldPath("bar.baz")
-                            .setSetToServerValue(
-                                DocumentTransform.FieldTransform.ServerValue.REQUEST_TIME)))
+            .setUpdate(Document.newBuilder().setName("projects/p/databases/d/documents/docs/1"))
+            .setUpdateMask(DocumentMask.newBuilder().build())
+            .addUpdateTransforms(
+                FieldTransform.newBuilder()
+                    .setFieldPath("a")
+                    .setSetToServerValue(ServerValue.REQUEST_TIME))
+            .addUpdateTransforms(
+                FieldTransform.newBuilder()
+                    .setFieldPath("bar.baz")
+                    .setSetToServerValue(ServerValue.REQUEST_TIME))
             .setCurrentDocument(Precondition.newBuilder().setExists(true))
             .build();
+    assertRoundTripForMutation(mutation, expected);
 
+    mutation =
```

Comment:
Super optional nit (here and below): To me, `set` is the base case and `patch` is more advanced. I would flip the order of these assertions.